owncloudData/src/main/java/com/owncloud/android/data/files/repository/OCFileRepository.kt

```diff
@@ -168,10 +168,14 @@ class OCFileRepository(
         val personalSpace = localSpacesDataSource.getPersonalSpaceForAccount(owner)
         if (personalSpace == null) {
             val legacyRootFolder = localFileDataSource.getFileByRemotePath(remotePath = ROOT_PATH, owner = owner, spaceId = null)
-            return legacyRootFolder!!
+            try {
+                return legacyRootFolder ?: throw IllegalStateException("LegacyRootFolder not found")
+            } catch (e: IllegalStateException) {
+                Timber.i("There was an error: $e")
```

Comment:
Placing this catch here could mute the error to the user.

Reply:
This comes from this CR: https://github.com/owncloud/android/pull/4170#discussion_r1338841950
It's true that the user won't receive any feedback, but legacyRootFolder should never be `null`, this allows us to catch the exception and prevent the app from crashing